### PR TITLE
[Fix] Fix  keypoint_scores in visualization

### PR DESCRIPTION
**Pull request by liqikai9**
## Motivation

1. Support nms-thr in `topdown_demo_with_mmdet.py`;
2. Fix use `keypoint_scores` to filter keypoints to be visualized.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmpose/visualization/local_visualizer.py [hunk 1]
```diff
@@ -225,8 +225,8 @@ def _draw_instances_kpts(self,
             keypoints = instances.get('transformed_keypoints',
                                       instances.keypoints)
 
-            if 'scores' in instances and self.show_keypoint_weight:
-                scores = instances.scores
+            if 'keypoint_scores' in instances:
```
Comment (Ben-Louis) [inline mmpose/visualization/local_visualizer.py:228]: Why is `self.show_keypoint_weight` omitted here?

File: demo/topdown_demo_with_mmdet.py [hunk 2]
```diff
@@ -98,13 +98,18 @@ def main():
         type=float,
         default=0.3,
         help='Bounding box score threshold')
+    parser.add_argument(
+        '--nms-thr',
+        type=float,
+        default=0.3,
+        help='Person detection score threshold for NMS')
```
Comment (Ben-Louis) [inline demo/topdown_demo_with_mmdet.py:105]:
```suggestion
        help='IoU threshold for bounding box NMS')
```

File: demo/topdown_face_demo.py [hunk 3]
```diff
@@ -97,13 +97,18 @@ def main():
         'Default not saving the visualization images.')
     parser.add_argument(
         '--device', default='cuda:0', help='Device used for inference')
+    parser.add_argument(
+        '--nms-thr',
+        type=float,
+        default=0.3,
+        help='Face detection score threshold for NMS')
```
Comment (Ben-Louis) [inline demo/topdown_face_demo.py:104]:
```suggestion
        help='IoU threshold for bounding box NMS')
```

File: mmpose/visualization/local_visualizer.py (hunk 1 above)
Comment (liqikai9) [reply mmpose/visualization/local_visualizer.py:228]: I think `keypoint_scores` shoud be irrevelant to `self.show_keypoint_weight`. 

A special case is that when I do not want to draw those keypoints with `keypoint_scores` lower than `kpt_score_thr`. If  `self.show_keypoint_weight = False`, and `keypoint_scores` is still present in instances, then although `keypoint_scores` is present in instances,  they are all forced to be assigned 1, and will be draw in the end.

Comment (Ben-Louis) [reply mmpose/visualization/local_visualizer.py:228]: Thanks, it makes sense. 

But `self.show_keypoint_weight` controls whether to plot keypoints with different transparency depending on keypoint_scores. When `self.show_keypoint_weight==False`, all visible keypoints should be plotted with the same transparency. This argument seems not to work now. To make it work, we also need to modify [this line](https://github.com/open-mmlab/mmpose/blob/e4549b13ea89dde824d5e4f90a7ef84c0364174d/mmpose/visualization/local_visualizer.py#L262).

Comment (liqikai9) [reply mmpose/visualization/local_visualizer.py:228]: How about this
```
if self.show_keypoint_weight:
    transparency = max(0, min(1, score[kid])) * self.alpha
```

Comment (Ben-Louis) [reply mmpose/visualization/local_visualizer.py:228]: `transparency` needs to be set when `self.show_keypoint_weight` is False
```python
transparency = self.alpha
if self.show_keypoint_weight:
    transparency *= max(0, min(1, score[kid]))  
```

Comment (liqikai9) [reply mmpose/visualization/local_visualizer.py:228]: Nice. I will fix it according to your advice.

**Review by Ben-Louis — APPROVED** [review]

**Pull request merged**